pkg/statsutil/stats_linux.go

```diff
@@ -0,0 +1,160 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package statsutil
+
+import (
+	"time"
+
+	v1 "github.com/containerd/cgroups/stats/v1"
+	v2 "github.com/containerd/cgroups/v2/stats"
+)
+
+var (
+	memPercent, cpuPercent float64
+	blkRead, blkWrite      uint64 // Only used on Linux
+	mem, memLimit          float64
+	netRx, netTx           float64
+	pidsStatsCurrent       uint64
+)
+
+func SetCgroupStatsFields(previousCgroupCPU, previousCgroupSystem uint64, data *v1.Metrics) (StatsEntry, error) {
+
+	cpuPercent = calculateCgroupCPUPercent(previousCgroupCPU, previousCgroupSystem, data)
+	blkRead, blkWrite = calculateCgroupBlockIO(data)
+	mem = calculateCgroupMemUsage(data)
+	memLimit = float64(data.Memory.Usage.Limit)
+	memPercent = calculateMemPercent(memLimit, mem)
+	pidsStatsCurrent = data.Pids.Current
+	netRx, netTx = calculateNetwork(data)
+
+	return StatsEntry{
+		CPUPercentage:    cpuPercent,
+		Memory:           mem,
+		MemoryPercentage: memPercent,
+		MemoryLimit:      memLimit,
+		NetworkRx:        netRx,
+		NetworkTx:        netTx,
+		BlockRead:        float64(blkRead),
+		BlockWrite:       float64(blkWrite),
+		PidsCurrent:      pidsStatsCurrent,
+	}, nil
+
+}
+
+func SetCgroup2StatsFields(previousCgroup2CPU, previousCgroup2System uint64, metrics *v2.Metrics) (StatsEntry, error) {
+
+	cpuPercent = calculateCgroup2CPUPercent(previousCgroup2CPU, previousCgroup2System, metrics)
+	blkRead, blkWrite = calculateCgroup2IO(metrics)
+	mem = calculateCgroup2MemUsage(metrics)
+	memLimit = float64(metrics.Memory.UsageLimit)
+	memPercent = calculateMemPercent(memLimit, mem)
+	pidsStatsCurrent = metrics.Pids.Current
```

Comment:
NetworkTx/Rx missing?

Reply:
@AkihiroSuda  it seems that network is not supported by cgroupv2 https://pkg.go.dev/github.com/containerd/cgroups@v1.0.2/v2/stats#Metrics

Reply:
But `docker stats` work with cgroup v2.
We can revisit this in another PR.

Reply:
@AkihiroSuda  I'm assuming that `docker` doesn't consume the cgroup api from containerd  https://github.com/containerd/cgroups. 
do you think it is a lack in the containerd cgroup api and that I have to open a ticket at  https://github.com/containerd/cgroups

Reply:
docker scans the stats from netlink, not from cgroups

Reply:
@AkihiroSuda thank you for checking this I will fix it.